app/controllers/reorder_announcements_controller.rb

```diff
@@ -6,7 +6,20 @@ def index
     coronavirus_page
   end
 
-  def update; end
+  def update
+    reordered_announcements = JSON.parse(params[:announcement_order_save])
+    reordered_announcements.each do |announcement_data|
+      announcement = coronavirus_page.announcements.find(announcement_data["id"])
+      announcement.update!(position: announcement_data["position"])
+    end
+
+    if draft_updater.send
+      redirect_to coronavirus_page_path(slug), notice: "Announcements were successfully reordered."
+    else
+      message = "Sorry! Announcements have not been reordered: #{draft_updater.errors.to_sentence}."
```

Comment:
Have error messages been put through content / do they need to?

Reply:
They haven't. I used the same errors/message format that sub-sections use. I figured they're fine as is as a starting point.

Reply:
nice one, good plan